### PR TITLE
Add C++ linter checks in Travis CI 

### PR DESCRIPTION
**Pull request by niboshi**
Adds C++ linter checks (clang-tidy, clang-format, cpplint) in Travis CI matrix.

Depends on: ~#5859~, ~#5868~, ~#5901~, ~#5933~

### CODE REVIEW

**Comment by hvy** [comment]
You can rebase now that https://github.com/chainer/chainer/pull/5859 is merged.

File: scripts/ci/travis/run-tests.sh [hunk 1]
```diff
@@ -19,56 +22,262 @@ esac
...
+}
+
+
+step_before_install_chainerx_style_check_deps() {
+    [ $TRAVIS_OS_NAME = "linux" ]  # currently only tested in linux
+
+    sudo apt-get install -y \
+         clang-format-6.0 \
+         parallel \
+
```
Comment (hvy) [inline scripts/ci/travis/run-tests.sh]:
```suggestion
```

Comment (niboshi) [reply scripts/ci/travis/run-tests.sh]: That would be a syntax error.

Comment (hvy) [reply scripts/ci/travis/run-tests.sh]: A minor but how about writing it on a single line?

File: scripts/ci/travis/run-tests.sh [hunk 2]
```diff
@@ -19,56 +22,262 @@ esac
...
-                    python --version;
+                    PYTHON_CONFIGURE_OPTS="--enable-unicode=ucs2" pyenv install -ks $PYTHON_VERSION
+                    pyenv global $PYTHON_VERSION
+                    python --version
 
-                    brew cask uninstall oclint;
-                    brew install hdf5;
-                    brew install open-mpi;
+                    brew cask uninstall oclint
+                    brew install hdf5
```
[the remark this answers is not in the document]
Comment (niboshi) [reply scripts/ci/travis/run-tests.sh]: Fixed.

File: scripts/ci/travis/run-tests.sh (hunk 1 above)
Comment (niboshi) [reply scripts/ci/travis/run-tests.sh]: Breaking into multiple lines like this is preferred.
* Easier to write comments on each item.
* Easier to understand it's a list.
* Diff will be minimized if partly modified.

Comment (hvy) [reply scripts/ci/travis/run-tests.sh]: Okay, let's keep the current code.

**Comment by hvy** [comment]
jenkins, test this please.

File: scripts/ci/travis/run-tests.sh [hunk 3]
```diff
@@ -19,56 +22,265 @@ esac
...
+    local reqs=(
+        theano
+        h5py
+        pillow
+    )
+    pip install "${reqs[@]}"
+}
+
+
+step_before_install_chainer_test() {
```
Comment (hvy) [inline scripts/ci/travis/run-tests.sh]: How about `_deps` suffix for consistency?

Comment (hvy) [reply scripts/ci/travis/run-tests.sh]: Discussed offline that it's not appropriate (as it does more than that).

**Comment by hvy** [comment]
LGTM!

**Pull request merged**